packages/bootstrap-chat/README.md

```diff
@@ -0,0 +1,141 @@
+
+
+## `<ChatApp />` documentation
+
+`ChatApp` exposes the following props: 
+
+All render props will receive as part of their args both `mephistoContext` and `appContext`, which will be described below.
+
+
+### `renderMessage({ message, idx, mephistoContext, appContext })`
+A render prop that if implemented, overides the way a chat message is rendered.
+
+**Arguments:**
+
+**`message`**: An object with the following schema:
+
+TODO: where does this object come from in the backend? Find to verify the below
+
+```
+{
+    message_id,
+    text,
+    task_data
+}
+```
+
+**`idx`**: The 0-based index of the current message in the list of all messages
+
+
+### `renderSidePane({ mephistoContext, appContext })`
+A render prop that if implemented, overrides the way the side pane is rendered.
+
+### `renderTextResponse({ onMessageSend, inputMode, active, mephistoContext, appContext })`
+A render prop that if implemented, overrides the way the response text box is rendered. The default implementation of `ChatApp` is rendering a `<TextResponse />` component here.
+
+Note: If `renderResponse()` is also implemented, this render prop will be ignored.
+
+This render prop is for cases where you'd like to minimally alter the response text box. For larger customizations, we recommend implementing `renderReponse()` instead.
+
+```js
+{
+    onMessageSend,
+    inputMode,
+    active, // helper boolean, essentially is true when inputMode is INPUT_MODE.READY_FOR_INPUT
+}
+```
+
+### `renderResponse({ onMessageSend, inputMode, mephistoContext, appContext })`
+A render prop that if implemented, overrides the way the entire response panel is rendered. You will be responsible for handling all aspects of the panel in this case.
+
+Note: If this render prop is implemented, `renderTextResponse()` will be ignored.
+
+
+### `onMessagesChange(messages)`
+
+A callback that will get called any time a new message is received. The callback will receive as an argument all messages received thus far.
+
+### The `mephistoContext` argument
+
+All render props will receive the `mephistoContext` argument.
+
+This object is basically the return value of calling the `useMephistoLiveTask` hooks. Further documentation on these properties can be found in the documentation for the `mephisto-task` package under the `useMephistoLiveTask` section.
+
+### The `appContext` argument
+
+All render props will receive the `appContext` argument. This context is generally used to handle app-specific information, as opposed to Mephisto-specific information.
```

Comment:
Nit - isn't `appContext` defined by `bootstrap-chat`, such that you could note all of the default arguments. (Also, is it possible to add arguments to appContext?)

Reply:
yep, we can definitely do this. I'll add it down below. Add the moment it's not possible to add more arguments, will also make a note of that